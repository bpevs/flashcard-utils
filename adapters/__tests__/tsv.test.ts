import { assertEquals } from 'jsr:@std/assert'
import { assertSnapshot } from 'jsr:@std/testing/snapshot'
import fromObj from '../from_obj.ts'
import fromTSV from '../from_tsv.ts'
import toTSV from '../to_tsv.ts'
import data from './__data__/zh_CN.ts'

/**
 * @todo Update TSV to support Anki-style TSV with headers
 * @reference https://docs.ankiweb.net/importing/text-files.html#file-headers
 */
Deno.test('init Deck to/from TSV', async (t) => {
  const deck = fromObj(data)
  const tsvResult = toTSV(deck)
  await assertSnapshot(t, tsvResult)
  const tsvDeck = fromTSV(tsvResult, {
    meta: {
      id: deck.id,
      name: deck.name,
      desc: deck.desc,
      key: deck.key,
      meta: deck.meta,
    },
  })

  assertEquals(deck.notes, tsvDeck.notes)
})
