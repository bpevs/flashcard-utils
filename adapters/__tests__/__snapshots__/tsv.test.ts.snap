export const snapshot = {};

snapshot[`init Deck to/from TSV 1`] = `
"category_en	category	emoji	text	text_en	pinyin
animal	动物	🐶	狗	dog	gǒu
animal	动物	🐈	猫	cat	māo
body	身体	🦶	脚	foot	jiǎo
"
`;
